### PR TITLE
Update podTemplate related docs

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

- Remove deprecated field from the documentation ; they are
  deprecated, let's not document their usage anymore.
- Add `Pod Template` documentation to `pipelinerun.md` as it was
  missing.

Would be nice to get this merged before tagging the 0.6 :angel: (so that the docs are good for 0.6)

/cc @dibyom @bobcatfish 

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) and [TaskRun](../tekton/publish-run.yaml) to build and release this image

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md)
are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes)
must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS)
and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes)
must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS),
and they must first be added
[in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

### CODE REVIEW

File: docs/pipelineruns.md [hunk 1]
```diff
@@ -132,6 +127,71 @@ spec:
...
+      taskRef:
+      name: myTask
+---
+apiVersion: tekton.dev/v1alpha1
+kind: PipelineRun
+metadata:
+  name: myPipelineRun
+spec:
+  pipelineRef:
+    name: yPipelinemyTask
```
Comment (sbwsg) [inline docs/pipelineruns.md]: `yPipelinemyTask` -> `myPipeline`?

Comment (vdemeester) [reply docs/pipelineruns.md]: oups =)

Comment (vdemeester) [reply docs/pipelineruns.md]: Updated :angel:

**Comment by bobcatfish** [comment]
Good call! Thanks @vdemeester !

**Pull request merged**